diff -aurNp port_uboot/cpu/arm920t/start.S u-boot-2009.11_tekkaman/cpu/arm920t/start.S
--- port_uboot/cpu/arm920t/start.S 2009-12-16 06:20:54.000000000 +0800
+++ u-boot-2009.11_tekkaman/cpu/arm920t/start.S 2010-03-28 17:16:12.000000000 +0800
@@ -114,8 +114,8 @@ start_code:
 orr r0, r0, #0xd3
 msr cpsr, r0

- bl coloured_LED_init
- bl red_LED_on
+@ bl coloured_LED_init
+@ bl red_LED_on

 #if defined(CONFIG_AT91RM9200DK) || defined(CONFIG_AT91RM9200EK)
 /* 
