--- lib_arm/board.c	2022-07-25 20:48:59.537520000 +1000
+++ ../../U-boot-2009.11_tekkaman/lib_arm/board.c	2022-07-25 21:40:13.517902674 +1000
@@ -49,6 +49,7 @@
 #include <nand.h>
 #include <onenand_uboot.h>
 #include <mmc.h>
+#include <s3c2410.h>    //tekkamanninja
 
 #ifdef CONFIG_BITBANGMII
 #include <miiphy.h>
@@ -86,7 +87,7 @@ extern void rtl8019_get_enetaddr (uchar
 #include <i2c.h>
 #endif
 
-
+#if 0
 /************************************************************************
  * Coloured LED functionality
  ************************************************************************
@@ -110,6 +111,7 @@ void inline __blue_LED_on(void) {}
 void blue_LED_on(void) __attribute__((weak, alias("__blue_LED_on")));
 void inline __blue_LED_off(void) {}
 void blue_LED_off(void) __attribute__((weak, alias("__blue_LED_off")));
+#endif
 
 /************************************************************************
  * Init Utilities							*
@@ -135,7 +137,13 @@ static int init_baudrate (void)
 
 static int display_banner (void)
 {
-	printf ("\n\n%s\n\n", version_string);
+#if defined(CONFIG_MINI2440_LED) 	
+	struct s3c24x0_gpio * const gpio = s3c24x0_get_base_gpio();
+	gpio->GPBDAT = 0x101; //tekkamanninja
+#endif
+	printf ("\n\n%s\n\n", version_string);	
+	printf (" modified by tekkamanninja (tekkamanninja@163.com)\n");
+	printf (" Love Linux forever!!\n\n");
 	debug ("U-Boot code: %08lX -> %08lX  BSS: -> %08lX\n",
 	       _armboot_start, _bss_start, _bss_end);
 #ifdef CONFIG_MODEM_SUPPORT
@@ -272,7 +280,9 @@ void start_armboot (void)
 #if defined(CONFIG_VFD) || defined(CONFIG_LCD)
 	unsigned long addr;
 #endif
-
+#if defined(CONFIG_MINI2440_LED) 	
+	struct s3c24x0_gpio * const gpio = s3c24x0_get_base_gpio();
+#endif
 	/* Pointer is writable since we allocated a register for it */
 	gd = (gd_t*)(_armboot_start - CONFIG_SYS_MALLOC_LEN - sizeof(gd_t));
 	/* compiler optimization barrier needed for GCC >= 3.4 */
@@ -434,6 +444,15 @@ extern void davinci_eth_set_mac_addr (co
 	reset_phy();
 #endif
 #endif
+#if defined(CONFIG_MINI2440_LED) 	
+		gpio->GPBDAT = 0x0; //tekkamanninja
+#endif
+ 
+#if defined(CONFIG_CFB_CONSOLE)        
+	printf ("%s\n", version_string);
+	printf ("modified by tekkamanninja\n(tekkamanninja@163.com)\n");
+	printf ("Love Linux forever!!\n");
+#endif
 	/* main_loop() can return to retry autoboot, if so just run it again. */
 	for (;;) {
 		main_loop ();
